src/wasm/wasm-type.cpp

```diff
@@ -392,6 +500,18 @@ std::ostream& operator<<(std::ostream& os, Type type) {
         os << "exnref";
         break;
     }
+  } else {
+    auto* typeDef = (TypeDef*)type.getID();
```

Comment:
It might be a readability win to have a small `getDef` function in this file to use rather than all these C-style casts.

```c++
static TypeDef* getDef(Type type) { return (TypeDef*)type.getID(); }
```